downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/MapOldStableIds.java

```diff
@@ -0,0 +1,118 @@
+package org.reactome.release.downloadDirectory;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.model.ReactomeJavaConstants;
+
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class MapOldStableIds {
+	private static final Logger logger = LogManager.getLogger();
+	
+	private static Connection connect = null;
+	private static Statement statement = null;
+	private PreparedStatement preparedStatement = null;
+	private static ResultSet resultSet = null;
+	
+	public static void execute(String username, String password, String host, int releaseNumber) throws Exception {
+		logger.info("Running MapOldStableIds");
+		// Need to use mysql driver to access stable_identifiers db
+		logger.info("Connecting to stable_identifers db...");
+		Class.forName("com.mysql.jdbc.Driver");
+		connect = DriverManager.getConnection("jdbc:mysql://" + host + "/stable_identifiers?" + "user=" + username + "&password=" + password);
+		statement = connect.createStatement();
+		resultSet = statement.executeQuery("SELECT identifier,instanceId FROM StableIdentifier");
+		
+		// Iterate through returned results of DB IDs and stable IDs 
+		logger.info("Mapping Old Stable IDs to Current Stable IDs...");
+		HashMap<String,ArrayList<String>> dbIdToStableIds = new HashMap<String,ArrayList<String>>();
+		ArrayList<String> dbIds = new ArrayList<String>();
+		while (resultSet.next()) {
+			String stableId = resultSet.getString(1);
+			String dbId = resultSet.getString(2);
+			
+			if (dbIdToStableIds.get(dbId) != null) {
+				dbIdToStableIds.get(dbId).add(stableId);
+			} else {
+				ArrayList<String> stableIds = new ArrayList<String>(Arrays.asList(stableId));
+				dbIdToStableIds.put(dbId, stableIds);
+				dbIds.add(dbId);
+			}
+		}
+		Collections.sort(dbIds);
+		
+		// Iterate through array of stable IDs associated with DB ID, splitting into human and non-human groups
+		ArrayList<ArrayList<Object>> hsaIds = new ArrayList<ArrayList<Object>>();
+		ArrayList<ArrayList<Object>> nonHsaIds = new ArrayList<ArrayList<Object>>();
+		for (String dbId : dbIds) {
+			ArrayList<String> stableIds = dbIdToStableIds.get(dbId);
+			Collections.sort(stableIds);
+
+			if (stableIds.size() < 2 || !stableIds.get(0).matches("^R-.*")) {
+				continue;
```

Comment:
Is there some other way to write this logic, than using a `continue` statement?

Reply:
More than likely. Is it bad practice to use `continue` or `break`?

Reply:
It's... debatable. You can read one of many threads here: https://softwareengineering.stackexchange.com/questions/58237/are-break-and-continue-bad-programming-practices

 I _personally_ don't like them, because as the accepted answer says:

> When used in the middle of the block, with some code around, they act like hidden traps, so it's bad.

But this does not mean they are forbidden, and there are arguments in favour of them on that thread as well. I would just say that if you are going to use them, make sure that you include some comments to make it 100% clear what is happening.

Personally, I try to rewrite such situations to use if-else. It sometimes results in longer code, but I think the trade-off of code that is quicker/easier to undertand and is more intuitive is worth it. But I also admit there have been times where I've used `break` and `continue`, so I'm not perfect. :P